api/api/controllers/search_controller.py

```diff
@@ -495,43 +496,42 @@ def search(
     return results, page_count, result_count, search_context.asdict()
 
 
-def related_media(uuid, index, filter_dead):
-    """Given a UUID, find related search results."""
+def related_media(uuid: str, index: str, filter_dead: bool) -> list[Hit]:
+    """Given a UUID, find related search results based on title and tags."""
 
-    search_client = Search(index=index)
+    # Search the default index for the item itself as it might be sensitive.
+    item_search = Search(index=index)
+    item_hit = item_search.query("match", identifier=uuid).execute().hits[0]
 
-    # Convert UUID to sequential ID.
-    item = search_client
-    item = item.query("match", identifier=uuid)
-    _id = item.execute().hits[0].id
+    # Match related using title.
+    title = item_hit.title
+    title_query = SimpleQueryString(query=title, fields=["title"])
+    related_query = title_query
 
-    s = search_client
-    s = s.query(
-        MoreLikeThis(
-            fields=["tags.name", "title", "creator"],
-            like={"_index": index, "_id": _id},
-            min_term_freq=1,
-            max_query_terms=50,
-        )
-    )
-    # Never show mature content in recommendations.
-    s = s.exclude("term", mature=True)
+    # Match related using tags, if the item has any.
+    if tags := getattr(item_hit, "tags", None):
+        # Only use the first 10 tags
+        tags = ",".join([tag.name for tag in tags[:10]])
+        tags_query = SimpleQueryString(fields=["tags.name"], query=tags)
+        related_query |= tags_query
```

Comment:
I wonder if a multi_match wouldn't be more straightforward than constructing a simple query string query? If we do use simple query string, though, `,` isn't an operator available, as far as I can tell: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html#simple-query-string-syntax

A space should be sufficient to use the default `OR` operator. In this case even `query_string` could work fine.